tools/functional-tester/etcd-runner/main.go

```diff
@@ -212,6 +219,173 @@ func runRacer(eps []string, round int) {
 	doRounds(rcs, round)
 }
 
+func runWatcher(eps []string) {
+	runningTime := 60 * time.Second // time for which operation should be performed
+	noOfPrefixes := 20              // total number of prefixes which will be watched upon
+	watchPerPrefix := 10            // number of watchers per prefix
+	reqRate := 20                   // put request per second
+	keyPrePrefix := 1000            // max number of keyPrePrefixs for put operation
+
+	// pre generate prefixes equal to noOfPrefixes
+	prefixes := generateUniqueKeys(5, noOfPrefixes)
+	// pre generate keys
+	prePrefixKeys := generateRandomKeys(10, keyPrePrefix)
+	// map for holding prefixes and values
+	prefixKeysMap := make(map[string][]string)
+	// actual keys for performing put operation
+	keys := make([]string, 0)
+
+	// initialize the map with all prefixes
+	for _, prefix := range prefixes {
```

Comment:
first a put key can trigger multiple watchers since we have N watchers on each prefix
second, if we want to a key to trigger different watchers on different prefix we can do this in a deterministic way. we can put keys `abc--` and make set of watchers watch on `abc-` and another set to watch on `abc--`.

moreover, the key point of the test is to test the watcher behavior under failure. we have not even got into that part yet. So, yes, i want to code and logic to be as simple as possible and as deterministic as possible. We probably need to debug it soon. That is why I suggest to pre generate everything and we should know exactly what is going to happen instead of determining the result on the flight.

making stuff complicated is just trying to create troubles for the future.